lib/src/ui/pages/profile/auth/authorization_notifier.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:allthenews/generated/l10n.dart';
+import 'package:allthenews/src/data/communication/firebase/firebase_authorization.dart';
+import 'package:allthenews/src/domain/authorization/authorization.dart';
+import 'package:allthenews/src/ui/pages/profile/auth/authorization_state.dart';
+import 'package:firebase_auth/firebase_auth.dart';
+import 'package:flutter/widgets.dart';
+
+class AuthorizationNotifier extends ChangeNotifier {
+  final Authorization authorization = FirebaseAuthorization();
+
+  AuthorizationState _state = const AuthorizationState();
+
+  AuthorizationState get state => _state;
+
+  VoidCallback returnToProfile;
+
+  void getCurrentUser() {
+    authorization.userStream().listen((user) {
+      _setNotifierState(AuthorizationState(user: user));
+    });
+  }
+
+  void createUser(String email, String password, String name) {
+    _setNotifierState(const AuthorizationState(isLoading: true));
+    authorization
+        .createUser(email, password)
+        .then((userCredentials) => _updateUserName(userCredentials.user, name))
+        .catchError((exception) => _onUserAuthFailed(exception));
+  }
+
+  void signIn(String email, String password) {
+    _setNotifierState(const AuthorizationState(isLoading: true));
+    authorization
+        .signIn(email, password)
+        .then((userCredentials) => returnToProfile.call())
+        .catchError((exception) => _onUserAuthFailed(exception));
+  }
+
+  void _updateUserName(User user, String name) {
+    authorization
+        .updateUser(name)
+        .then((_) => returnToProfile.call())
```

Comment:
`returnToProfile?.call()` i w pozostałych miejscach też z `?`

Reply:
no właśnie zastanawiałem się czy to konieczne, dałem z `?` przy logoucie bo to jest na ekranie profilu i z tamtego miejsca nie przekazujemy callbacku do cofania. Pozostałe dwa miejsca przy metodach 'signIn` i `_updateUserName` będą wywołane tylko po wejściu do erkanu logowania/rejestracji, więc ten callback nie powinien być tam nullem 🤔

Reply:
a poza tym to myślałem czy warto na tym etapie przerzucać się na nawigation 2.0, co myślisz?

Reply:
powiedzmy, że z poziomu samego notifiera widzisz tylko, że callback może być zainicjalizowany ale nie musi, bo nie jest to realizowane przez konstruktor, więc w przypadku kiedy zawsze będzie `?` nie ma potrzeby zastanawiania się czy ma prawo wystąpić czy nie, bo w zależności od użycia może ;) Co do navigation 2.0 to jak najbardziej :)

Reply:
ma to sens, dodam `?`. Dodałem też taska na trello do navigatora :)